lib/3scale/backend/stats/cleaner.rb

```diff
@@ -76,12 +76,31 @@ def delete!(redis_conns, log_deleted_keys: nil)
             services = services_to_delete
             logger.info("Going to delete the stats keys for these services: #{services.to_a}")
 
+            delete_successful = true
             unless services.empty?
               redis_conns.each do |redis_conn|
-                delete_keys(redis_conn, services, log_deleted_keys)
+                begin
+                  delete_keys(redis_conn, services, log_deleted_keys)
+                # If it's a connection error, mark as failed and continue
+                # cleaning other shards. If it's another kind of error, it
+                # could be a bug, so better re-raise.
+                rescue Redis::BaseConnectionError, Errno::ECONNREFUSED, Errno::EPIPE
+                  delete_successful = false
```

Comment:
can't replace with `find_all` + `any?`/`all?`?

Reply:
Not sure what you mean.

Reply:
I guess you meant something like `delete_successful = redis_conns.all? ... `

Reply:
yes

Reply:
It's a bit better. Changed :+1:

Reply:
In the end, I left it as it was at the beginning. We cannot use `all?` because it stops when there's a false, and I think that `find_all` + `any?` is more complex than the original solution.